include/nbl/builtin/hlsl/workgroup/arithmetic.hlsl

```diff
@@ -15,110 +16,107 @@ namespace hlsl
 namespace workgroup
 {
 
-template<typename T, class Binop, class ScratchAccessor>
+template<typename T, class Binop, class SharedAccessor>
 struct reduction
 {
-    struct inclusive_scan_t : subgroup::inclusive_scan<T, Binop, ScratchAccessor, false> {}; // Yes, inclusive scan subgroup op is used for reduction workgroup ops
+    struct inclusive_scan_t : subgroup::inclusive_scan<T, Binop> {}; // Yes, inclusive scan subgroup op is used for reduction workgroup ops
     T operator()(T value)
     {
-		Barrier();
-		Binop op;
-        WorkgroupScanHead<T, inclusive_scan_t, ScratchAccessor> wsh = WorkgroupScanHead<T, inclusive_scan_t, ScratchAccessor>::create(false, op.identity(), _NBL_HLSL_WORKGROUP_SIZE_);
+		glsl::barrier();
+        WorkgroupScanHead<T, inclusive_scan_t, SharedAccessor> wsh = WorkgroupScanHead<T, inclusive_scan_t, SharedAccessor>::create(false, Binop::identity(), _NBL_HLSL_WORKGROUP_SIZE_);
         T result = wsh(value);
-        Barrier();
-        T retVal = Broadcast<uint, ScratchAccessor>(result, wsh.lastInvocationInLevel);
-		Barrier();
-		MemoryBarrierShared();
+        glsl::barrier();
+        T retVal = Broadcast<uint, SharedAccessor>(result, wsh.lastInvocationInLevel);
+		glsl::barrier();
+		glsl::memoryBarrierShared();
```

Comment:
use the accessors workgroup and memoyr barrier method that you've just defined

Reply:
always use the `SharedAccessor::workgroupAndMemoryBarrier` instead of any `glsl::barrier()`